components/mediation/security/org.wso2.micro.integrator.mediation.security/src/main/java/org/wso2/micro/integrator/mediation/security/vault/SecretCipherHander.java

```diff
@@ -46,4 +80,31 @@ String getSecret(String alias) {
 		return parentRepository.getSecret(alias);
 	}
 
-}
\ No newline at end of file
+	public String getSecret(String alias, SecretSrcData secretSrcData) {
+		switch (secretSrcData.getVaultType()) {
+			case DOCKER:
+				// resolve path and alias
+				String resolvedDockerAlias = DOCKER_SECRET_ROOT + alias;
+				// For docker type we support plaintext as well
+				if (secretSrcData.isEncrypted()) {
+					return fileSecretRepository.getSecret(resolvedDockerAlias);
+				}
+				return fileSecretRepository.getPlainTextSecret(resolvedDockerAlias);
```

Comment:
Can we move this logic inside the fileSecretRepository itself. same goes with the other types of secreteRepositories. I think we can remove a lot of duplication if we do that. WDYT?